### PR TITLE
test: mining_getblocktemplate_longpoll.py improvements (use MiniWallet, add logging)

### PR DESCRIPTION
**Pull request by theStack**
This PR enables one more of the non-wallet functional tests (mining_getblocktemplate_longpoll.py) to be run even with the Bitcoin Core wallet disabled by using the new MiniWallet instead, as proposed in #20078. Also adds missing log messages for the subtests.

This was the only functional test that used the `random_transaction` helper in `test_framework/util.py`, hence it is removed, together with other helpers (`make_change` and `gather_inputs`) that were again only used by `random_transaction`.

### CODE REVIEW

File: test/functional/mining_getblocktemplate_longpoll.py [hunk 1]
```diff
@@ -29,45 +31,48 @@ def set_test_params(self):
         self.num_nodes = 2
         self.supports_cli = False
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
-        self.log.info("Warning: this test will take about 70 seconds in the best case. Be patient.")
-        self.nodes[0].generate(10)
+        self.log.info("Warning: this test will take about 70 seconds in the worst case. Be patient.")
```
Comment (MarcoFalke) [inline test/functional/mining_getblocktemplate_longpoll.py:35]: why this change?

File: test/functional/mining_getblocktemplate_longpoll.py [hunk 2]
```diff
@@ -29,45 +31,48 @@ def set_test_params(self):
...
+        self.nodes[1].generate(100)
+
+        self.log.info("Test that introducing a new transaction into the mempool will terminate the longpoll")
         thr = LongpollThread(self.nodes[0])
         thr.start()
         # generate a random transaction and submit it
         min_relay_fee = self.nodes[0].getnetworkinfo()["relayfee"]
-        # min_relay_fee is fee per 1000 bytes, which should be more than enough.
-        (txid, txhex, fee) = random_transaction(self.nodes, Decimal("1.1"), min_relay_fee, Decimal("0.001"), 20)
+        fee_rate = min_relay_fee + Decimal('0.00000010') * random.randint(0,20)
```
Comment (MarcoFalke) [inline test/functional/mining_getblocktemplate_longpoll.py:73]: Can remove unused `random_transaction`?

File: test/functional/mining_getblocktemplate_longpoll.py [hunk 3]
```diff
@@ -29,45 +31,48 @@ def set_test_params(self):
...
         thr.start()
-        self.nodes[0].generate(1)  # generate a block on another node
+        miniwallets[0].generate(1)  # generate a block on own node
         thr.join(5)  # wait 5 seconds or until thread exits
         assert not thr.is_alive()
 
-        # Test 4: test that introducing a new transaction into the mempool will terminate the longpoll
+        # Add enough mature utxos to the wallets, so that all txs spend confirmed coins
+        self.nodes[0].generate(100)
+        self.nodes[1].generate(100)
```
Comment (MarcoFalke) [inline test/functional/mining_getblocktemplate_longpoll.py:66]: Why are 100 not enough?

File: test/functional/mining_getblocktemplate_longpoll.py (hunk 1 above)
Comment (theStack) [reply test/functional/mining_getblocktemplate_longpoll.py]: Hm, I had a few test runs that finished in ~6 seconds (i.e. introduction of a new transaction would terminate the longpoll immediately, not only in +1minute), but can't reproduce it anymore. Changed the message back to original.

File: test/functional/mining_getblocktemplate_longpoll.py (hunk 2 above)
Comment (theStack) [reply test/functional/mining_getblocktemplate_longpoll.py:73]: Nice catch, thanks. Added a commit that removes the helpers `random_transaction` as well as `make_change` and `gather_inputs`. The latter two were only used by `random_transaction`.

File: test/functional/mining_getblocktemplate_longpoll.py (hunk 3 above)
Comment (theStack) [reply test/functional/mining_getblocktemplate_longpoll.py]: The intention was to ensure that both nodes have matured utxos, not only the one where we generate the blocks. However, only generating on one node and simply adding a `self.sync_blocks()` is the better solution. Fixed.

**Comment by theStack** [comment]
Updated PR description and force-pushed with the feedback from MarcoFalke taken into account.

File: test/functional/mining_getblocktemplate_longpoll.py [hunk 4]
```diff
@@ -29,45 +31,48 @@ def set_test_params(self):
...
+        self.sync_blocks()
+
+        self.log.info("Test that introducing a new transaction into the mempool will terminate the longpoll")
         thr = LongpollThread(self.nodes[0])
         thr.start()
         # generate a random transaction and submit it
         min_relay_fee = self.nodes[0].getnetworkinfo()["relayfee"]
-        # min_relay_fee is fee per 1000 bytes, which should be more than enough.
-        (txid, txhex, fee) = random_transaction(self.nodes, Decimal("1.1"), min_relay_fee, Decimal("0.001"), 20)
+        fee_rate = min_relay_fee + Decimal('0.00000010') * random.randint(0,20)
```
Comment (MarcoFalke) [inline test/functional/mining_getblocktemplate_longpoll.py:73]: why does the fee need to be random?

**Review by maflcko — APPROVED** [review]

**Comment by MarcoFalke** [comment]
ACK b128b566725a5037fdaea99940d1b9de5553d198

File: test/functional/mining_getblocktemplate_longpoll.py (hunk 4 above)
Comment (theStack) [reply test/functional/mining_getblocktemplate_longpoll.py:73]: My idea was to keep the test as close as possible to the original and only tackle the parts involving the wallet. Agree though that a fixed fee would work as well.
